### ISSUE TITLE
SOLC not installed error is not clear

### ISSUE THREAD

**Issue by feliam** [trailofbits/manticore#867]
2018-04-12 16:07:23,054: [9345] m.main:INFO: Beginning analysis
Traceback (most recent call last):
 File "/usr/local/bin/manticore", line 9, in <module>
   load_entry_point('manticore==0.1.8', 'console_scripts', 'manticore')()
 File "/usr/local/lib/python2.7/dist-packages/manticore/__main__.py", line 110, in main
   ethereum_cli(args)
 File "/usr/local/lib/python2.7/dist-packages/manticore/__main__.py", line 99, in ethereum_cli
   m.multi_tx_analysis(args.argv[0], args.contract, args.txlimit, args.txaccount)
 File "/usr/local/lib/python2.7/dist-packages/manticore/ethereum.py", line 929, in multi_tx_analysis
   contract_account = self.solidity_create_contract(source_code, contract_name=contract_name, owner=owner_account)
 File "/usr/local/lib/python2.7/dist-packages/manticore/ethereum.py", line 825, in solidity_create_contract
   compile_results = self._compile(source_code, contract_name)
 File "/usr/local/lib/python2.7/dist-packages/manticore/ethereum.py", line 616, in _compile
   p = Popen([solc, '--combined-json', 'abi,srcmap,srcmap-runtime,bin,hashes,bin-runtime', '--allow-paths', '.', temp.name], stdout=PIPE, stderr=PIPE)
 File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
   errread, errwrite)
 File "/usr/lib/python2.7/subprocess.py", line 1327, in _execute_child
   raise child_exception
OSError: [Errno 2] No such file or directory

**Comment by disconnect3d** [comment]
Might be good to have something similar to https://github.com/pwndbg/pwndbg/blob/dev/pwndbg/which.py#L42

We could change it to raise `OSError` if the program is not found instead of `None`.

Then we can have some module like `paths.py` (not sure if it is the best name) which will just do e.g.:
```python
solc = which('solc')
```

So when we will import `solc` it will check if it exists.

---
On the other hand it introduces [TOCTOU problem here](https://en.wikipedia.org/wiki/Time_of_check_to_time_of_use) but I don't really think it is that big matter.

**Comment by Martin2877** [comment]
how to install solc and add it to $PATH?

**Comment by Martin2877** [comment]
I fix it after:
```
sudo add-apt-repository ppa:ethereum/ethereum

sudo apt-get update

sudo apt-get install solc
```

**Comment by mossberg** [comment]
closing as this seems to be fixed in #877
